gpkit/constraints/set.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, constraints, substitutions=None):
                         # so we can catch them (see above) in ConstraintSets
             if hasattr(self[i], "substitutions"):
                 self.substitutions.update(self[i].substitutions)
+                self[i].substitutions = self.substitutions
```

Comment:
1. oooh, duh. Got it.

2. I know, it's a good improvement, and much more understandable -- I'm just saying that it's still not documented, and I think users still find this confusing and would appreciate a clear explanation of how it works. To be clear, I totally agree with doing the mutation.